### PR TITLE
Feature/it supervision

### LINKED ISSUE #69: Feature IT supervision
Branch to add the IT supervision application

- Technology name : it_supervision
- Version : 0.1

We want to be able to monitor a Saagie plateforme through several KPI:
- Saagie jobs / pipeline
- HDFS storage

Yes

### PR DESCRIPTION
**Pull request by CourbeB**
close #69

### CODE REVIEW

**Review by tquiviger — COMMENTED** [review]
Why three levels of directories (saagie-monitoring-tool > saagie-monitoring-tool > saagie-monitoring-tool)?

File: technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+FROM grafana/grafana:8.0.6-ubuntu
```
Comment (tquiviger) [inline technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile]: Can we use the latest Grafana version? 8.2.4

File: technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+RUN mkdir /opt/grafana && mkdir /opt/plugins && mkdir /app
+
+RUN apt-get update
+
+# Install Dependencies
+RUN apt-get update \
+  && apt-get install -y software-properties-common nginx cron wget libpq-dev python-dev openjdk-11-jdk ca-certificates-java \
+  && add-apt-repository ppa:deadsnakes/ppa \
+  && apt-get update \
+  && apt-get install -y python3.7 pip \
```
Comment (tquiviger) [inline technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile]: Can we use a more recent version of Python?

**Comment by CourbeB** [comment]
> Why three levels of directories (saagie-monitoring-tool > saagie-monitoring-tool > saagie-monitoring-tool)?

fixed

File: technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ENV JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64/
+ENV HADOOP_HOME=/hadoop/hadoop-2.6.5
+ENV HADOOP_CONF_DIR=/hadoop/hadoop-2.6.5/etc/hadoop
+ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/hadoop/hadoop-2.6.5/lib/native:/usr/lib/jvm/java-11-openjdk-amd64/lib"
+ENV CLASSPATH="/etc/hadoop/conf:/hadoop/hadoop-2.6.5/share/hadoop/common/lib/*:/hadoop/hadoop-2.6.5/share/hadoop/common/*:/hadoop/hadoop-2.6.5/share/hadoop/hdfs:/hadoop/hadoop-2.6.5/share/hadoop/hdfs/lib/*:/hadoop/hadoop-2.6.5/share/hadoop/hdfs/*:/hadoop/hadoop-2.6.5/share/hadoop/yarn/lib/*:/hadoop/hadoop-2.6.5/share/hadoop/yarn/*:/hadoop/hadoop-2.6.5/share/hadoop/mapreduce/lib/*:/hadoop/hadoop-2.6.5/share/hadoop/mapreduce/*"
+ENV PATH "/hadoop/hadoop-2.6.5/bin:${PATH}"
+
+# Install PostgreSQL
+# check for default search text language (now english, swithc to french?)
+RUN apt-get update \
```
Comment (tquiviger) [inline technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile]: maybe install postgresql before, when installing other libs with apt?

File: technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+ENV CRON_DAYS=1
+
+USER postgres
+RUN /usr/lib/postgresql/12/bin/initdb -D /var/lib/postgresql/data
+
+USER root
+COPY infra.sql infra.sql
+
+ADD grafana/provisioning /etc/grafana/provisioning
```
Comment (tquiviger) [inline technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile]: there is a mix of COPY and ADD, maybe just use one type of operator?

File: technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+ADD grafana/dashboards /var/lib/grafana/dashboards
+
+ENV GF_PATHS_DATA /opt/grafana
+ENV GF_PATHS_PLUGINS /opt/plugins
+
+RUN echo "export PATH=$PATH:/usr/lib/postgresql/12/bin" >> /etc/profile && . /etc/profile
+ADD code /app
+RUN pip install -r /app/requirements.txt
```
Comment (tquiviger) [inline technologies/app/saagie-monitoring-tool/saagie-monitoring-tool/saagie-monitoring-tool/Dockerfile]: can we put it before? whenever I update a config in Grafana, this requirements.txt is launched again

**Review by tquiviger — APPROVED** [review]

**Pull request merged**